pkg/report/reporter_test.go

```diff
@@ -0,0 +1,83 @@
+/*
+ * @license
+ * Copyright 2024 Dynatrace LLC
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package report_test
+
+import (
+	"context"
+	"errors"
+	"testing"
+	"time"
+
+	"github.com/spf13/afero"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/dynatrace/dynatrace-configuration-as-code/v2/pkg/config/coordinate"
+	"github.com/dynatrace/dynatrace-configuration-as-code/v2/pkg/report"
+)
+
+type testClock struct{ t time.Time }
+
+func (c *testClock) Now() time.Time { return c.t }
+
+// TestReporter_ContextWithNoReporterDiscards tests that the Recorder obtained from an context without the default one discards.
+func TestReporter_ContextWithNoReporterDiscards(t *testing.T) {
+	ctx := context.TODO()
+	reporter := report.GetReporterFromContextOrDiscard(ctx)
+	require.NotNil(t, reporter)
+
+	reporter.ReportDeployment(coordinate.Coordinate{Project: "test", Type: "dashboard", ConfigId: "my-dashboard"}, report.State_DEPL_SUCCESS, nil, nil)
+	reporter.Stop()
+	assert.Empty(t, reporter.GetSummary(), "discarding Reporter should not return a summary")
+}
+
+// TestReporter_ContextWithDefaultReporterCollectsEvents tests that the Reporter obtained from an context with the default one attached collects events.
+func TestReporter_ContextWithDefaultReporterCollectsEvents(t *testing.T) {
+
+	reportFilename := "test_report.jsonl"
+	fs := &afero.MemMapFs{}
```

Comment:
nitpick: Please don't use the MemMapFs if easily possible. The reason is, that we had Windows bugs in the past that we did not catch because we used Memory mapping instead of the 'real' paths. It would be awesome if you could use the internal/testutils/afero_test_utils.go:37#TempFs function, which uses a temp directory instead and will create/read the actual files.